coil-video/src/main/java/coil/decode/VideoFrameDecoder.kt

```diff
@@ -35,7 +40,13 @@ class VideoFrameDecoder(
     override suspend fun decode() = MediaMetadataRetriever().use { retriever ->
         retriever.setDataSource(source)
         val option = options.parameters.videoFrameOption() ?: OPTION_CLOSEST_SYNC
-        val frameMicros = options.parameters.videoFrameMicros() ?: 0L
+
+        val frameMicros = options.parameters.videoFrameMicros()
+            ?: options.parameters.videoFramePercent()?.let { framePercent ->
+                val durationMillis = retriever.extractMetadata(METADATA_KEY_DURATION)
+                    ?.toLongOrNull() ?: 0L
+                MICROSECONDS.convert((framePercent * durationMillis).roundToLong(), MILLISECONDS)
```

Comment:
Oops - looks like this should be the other way around. I'll correct it on `main` + add a test.

Reply:
Why? [`TimeUnit.convert()`](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/TimeUnit.html#convert-long-java.util.concurrent.TimeUnit-) takes `long sourceDuration, TimeUnit sourceUnit`, and we are indeed converting from milliseconds to microseconds here.

Reply:
Ah sorry my mistake! Either way I've added a regression test on `main` to verify the behaviour.